components/workspace/Panel.tsx

```diff
@@ -63,7 +70,37 @@ export function WorkspacePanel(props: PropsOf<typeof ActionPanel>) {
       .getAll('workspaceItem')
       .filter((item): item is string => typeof item === 'string')
 
-    const dids = selectedItems.filter((item) => item.startsWith('did:'))
+    // For every selected item, find out which DID it corresponds
+    const dids = selectedItems
+      .map((item) => {
+        if (item.startsWith('did:')) return item
+
+        const status = workspaceListStatuses?.[item]
+
+        if (ToolsOzoneModerationDefs.isRepoViewDetail(status)) {
+          return status.did
+        }
+
+        if (ToolsOzoneModerationDefs.isRecordViewDetail(status)) {
+          return status.repo.did
+        }
+
+        if (ToolsOzoneModerationDefs.isSubjectStatusView(status)) {
+          const { subject } = status
+          if (ComAtprotoAdminDefs.isRepoRef(subject)) {
+            return subject.did
+          }
+
+          if (ComAtprotoRepoStrongRef.isMain(subject)) {
+            // Don't know
+            return undefined
```

Comment:
we can still get the did from subject.uri in this case, right? `new AtUri(subject.uri).host` should give you that I think